src/capi.rs

```diff
@@ -97,19 +101,25 @@ pub unsafe extern "C" fn microvmi_read_physical(
     size: size_t,
     bytes_read: *mut uint64_t,
 ) -> bool {
-    let driver = get_driver_mut_ptr(context);
+    let mut driver = get_driver_mut_ptr(context);
 
-    let mut bytes_read_local = 0;
-    let res = (*driver)
-        .read_physical(
-            physical_address,
-            slice::from_raw_parts_mut(buffer, size),
-            &mut bytes_read_local,
-        )
-        .is_ok();
+    if driver
```

Comment:
This file was supposed to be a simple C abstraction layer. Augmenting it with code that goes beyond that is bad design imho.